x-pack/solutions/security/plugins/security_solution/server/ui_settings.ts

```diff
@@ -512,3 +514,29 @@ export const initUiSettings = (
 
   uiSettings.register(orderSettings(securityUiSettings));
 };
+export const getDefaultAIConnectorSetting = (connectors: Connector[]): SettingsConfig | null =>
+  connectors.length > 0
+    ? {
+        [DEFAULT_AI_CONNECTOR]: {
+          name: i18n.translate('xpack.securitySolution.uiSettings.defaultAIConnectorLabel', {
+            defaultMessage: 'Default AI Connector',
+          }),
+          // TODO, make Elastic LLM the default value once fully available in serverless
+          value: connectors[0].id,
+          description: i18n.translate(
+            'xpack.securitySolution.uiSettings.defaultAIConnectorDescription',
+            {
+              // TODO update this copy, waiting on James Spiteri's input
+              defaultMessage: 'Default AI connector for serverless AI features (AI for SOC)',
+            }
+          ),
+          type: 'select',
+          options: connectors.map(({ id }) => id),
```

Comment:
What happens if the user sets a connector here and then it is removed? Will this setting become invalid the next time it's opened? Just curious

Reply:
@stephmilovic I don't know how to do it. I am not sure if there are more settings defined in a "dynamic" way, it seems the rest of the UI settings are defined in a static way. I think we should talk with the code owners about this issue.

Reply:
I've got it! check this commit: https://github.com/elastic/kibana/pull/218018/commits/0f10c5aac8e33a4011833b11d2661a4e4318a685

https://github.com/user-attachments/assets/12fb69d0-c504-418c-8e14-854f95e8577f

Can I please get review from @elastic/kibana-management for this change?

Reply:
Oh nice! That's a smart fix 👍